### PR TITLE
[Spree Upgrade] Fix taxes tests in models/spree/order_spec

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Continues #2685 

#### What should we test?
Taxes tests (6 tests) should be fixed in order_spec.

### CODE REVIEW

File: spec/models/spree/order_spec.rb [hunk 1]
```diff
@@ -287,19 +288,23 @@
...
 
     before do
-      shipment.add_shipping_method(shipping_method, true)
       Spree::Config.shipment_inc_vat = true
       Spree::Config.shipping_tax_rate = tax_rate15.amount
-      Spree::TaxRate.adjust(order)
+
+      # This triggers the copy of the tax_category from product to line_item
+      #   The line item tax category is the one used for the calculation
+      line_item.save
```
[the remark this answers is not in the document]
Comment (mkllnk) [reply spec/models/spree/order_spec.rb]: I think the problem is that it has not been created properly. So the save copies the tax category which wasn't inititalised correctly.

[the remark this answers is not in the document]
Comment (luisramos0) [reply spec/models/spree/order_spec.rb]: great, thanks for this heads up. I revisited this and concluded I only needed to reload the line_item or assign it to the order, and the hook gets triggered with that

[the remark this answers is not in the document]
Comment (luisramos0) [reply spec/models/spree/order_spec.rb]: I pushed a new version

[the remark this answers is not in the document]
Comment (luisramos0) [reply spec/models/spree/order_spec.rb]: fyi, I have used this to improve #2812  (adjustment spec) as well. Thanks!

**Comment by luisramos0** [comment]
ready for re-review ;-)

File: spec/models/spree/order_spec.rb [hunk 2]
```diff
@@ -259,11 +259,12 @@
     let(:shipment) { create(:shipment_with, :shipping_method, shipping_method: shipping_method) }
     let(:order)          { create(:order, shipments: [shipment]) }
     let(:enterprise_fee) { create(:enterprise_fee) }
-    let!(:adjustment)    { create(:adjustment, adjustable: order, originator: enterprise_fee, label: "EF", amount: 123, included_tax: 2) }
 
     before do
       Spree::Config.shipment_inc_vat = true
       Spree::Config.shipping_tax_rate = 0.25
+
+      adjustment = create(:adjustment, adjustable: order, originator: enterprise_fee, label: "EF", amount: 123, included_tax: 2)
```
Comment (mkllnk) [inline spec/models/spree/order_spec.rb]: Do you need to assign it to the variable `adjustment`?

Comment (luisramos0) [reply spec/models/spree/order_spec.rb]: ah, no. good point! I removed it. :+1:

**Pull request merged**